rustls/src/server/tls13.rs

```diff
@@ -1276,6 +1272,7 @@ impl State<ServerConnectionData> for ExpectFinished {
         // Application data may now flow, even if we have client auth enabled.
         cx.common
             .start_traffic(&mut cx.sendable_plaintext);
+        cx.common.peer_identity = self.peer_identity;
```

Comment:
While I think minimizing writes into `CommonState` (particularly as a replacement for maintaining handshake state) is a good idea, IMO delaying this should not be a goal in itself. Rather, I imagine at a lower-level of abstraction we would yeet the peer's identity to the caller as soon as we don't need it for the handshake itself anymore.

Reply:
Yeah, but the identity can't change after we have established/received it, right?

Reply:
True, it cannot change. I think my point is that surfacing it as soon as it and a valid signature by it is received is not great (in TLS1.2 without EMS) because the amount of handshake covered by the signature isn't enough. That means an application that observes the certificate before the finished message is validated has the potential to be misled. It's a pretty marginal point.